tests/tests.cpp

```diff
@@ -2034,7 +2031,7 @@ TEST_F(EngineBasicTest, TestBackGroundCleaner) {
   {
     std::vector<std::thread> ts;
     ts.emplace_back(std::thread(SetString, 0));
```

Comment:
If we don't really need those ids, don't add them. Other reviewers may expect those ids fill a special purpose, such as to establish dependency between threads, but they don't.
Please keep things as simple as possible.